alica_engine/src/engine/teammanager/TeamManager.cpp

```diff
@@ -53,12 +53,13 @@ bool AgentsCache::addAgent(Agent* agent)
     return ret.second;
 }
 
-TeamManager::TeamManager(AlicaEngine* engine)
+TeamManager::TeamManager(AlicaEngine* engine, AgentIDConstPtr agentID)
         : _localAgent(nullptr)
         , _engine(engine)
         , _agentAnnouncementTimeInterval(AlicaTime::zero())
         , _timeLastAnnouncement(AlicaTime::zero())
         , _announcementRetries(0)
+        , _agentId(agentID)
```

Comment:
don't need to store it, the member Agent* _localAgent will already have the id once constructor finishes execution.
we can pass it to readselffromconfig() as an argument